File: Q02/main.c
#include<stdio.h>
int main(){
    int n;
    scanf("%d",&n);

    int arr[n];
    for(int i=0;i<n;i++){
        scanf("%d",&arr[i]);
    }

    int start=0,end=n-1;
    while(start<end){
        if(arr[start]!=arr[end]){
            printf("Not Palindrome");
            return 0;
        }
        start++;end--;
    }
    printf("Palindrome");
    return 0;
}

File: Q02/main.py
n=int(input())
arr=[]
for i in range(n):
    arr.append(int(input()))
start,end=0,n-1
while start<end:
    if arr[start]!=arr[end]:
        print("Not Palindrome")
        break
    start+=1
    end-=1
else:
    print("Palindrome")